src/sentry/servermode.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+
+import abc
+import itertools
+from enum import Enum
+from typing import Any, Callable, Iterable
+
+from django.conf import settings
+
+
+class ServerComponentMode(Enum):
+    """Defines the component mode of the application to be acting as."""
+
+    MONOLITH = "MONOLITH"
+    CONTROL = "CONTROL"
+    CUSTOMER = "CUSTOMER"
+    FRONTEND = "FRONTEND"
+
+    @classmethod
+    def resolve(cls, mode: str | ServerComponentMode | None) -> ServerComponentMode:
+        if not mode:
+            return cls.MONOLITH
+        if isinstance(mode, ServerComponentMode):
+            return mode
+
+        try:
+            obj = getattr(cls, mode)
```

Comment:
Should we use `server_mode` or `application_mode` to avoid future conflicts with fields on models named `mode`?

Reply:
`mode` isn't an attribute name. This is a minor kludge to resolve, e.g., the string `"CONTROL"` into the value `ServerComponentMode.CONTROL`.

Reply:
Turns out there's a way to do it without `getattr`: https://github.com/getsentry/sentry/pull/35994/commits/0ce9bee203707b266faf25711f6a9860fba00030

(Which makes sense, but was none too obvious. 😠 Thanks for the nudge that made me Google it. 😅)